Doc update (#123)
ROCmSoftwarePlatform/TransferBench: docs/install/install.rst

.. meta::
  :description: TransferBench is a utility to benchmark simultaneous transfers between user-specified devices (CPUs or GPUs)
  :keywords: Build TransferBench, Install TransferBench, API, ROCm, HIP

.. _install-transferbench:

---------------------------
Installing TransferBench
---------------------------

> This document provides information required to install and build TransferBench.

Reviewer: ```suggestion
This topic describes how to install and build TransferBench.
```

Reviewer: is it build and install or the other way around?

Author: Done.

Prerequisite
---------------

> * Install ROCm stack on the system to obtain HIP runtime

Reviewer: HIP should be hyperlinked to the HIP docs. Perhaps a very short description would be good here, too

Author: Added hyperlink

.. meta::
  :description: TransferBench documentation
  :keywords: TransferBench, API, ROCm, HIP

.. _install-transferbench:

---------------------------

> Installation

Reviewer: ```suggestion
Installing TransferBench
```

Reviewer: The titles should not be ambiguous and meet our SEO standards

---------------------------

This document provides information required to install and build TransferBench.

Prerequisite
---------------

> * Install ROCm stack on the system (HIP runtime)

Reviewer: Add something like: For more information, see HIP. 

This line is hard to understand. Do you need both ROCm and the HIP component installed?

Author: HIP runtime is installed when u install ROCm. Changed it to "Install ROCm stack on the system to obtain HIP runtime" to make this explicit.

* Install ``libnuma`` on the system

> * Enable AMD IOMMU and set to passthrough for AMD Instinct cards

Reviewer: ```suggestion
* Enable AMD IOMMU and set to passthrough for AMD Instinct accelerators
```

Reviewer: Should we provide an xref for more info here?

Author: @gilbertlee-amd which one of these links can we use here? 
https://rocm.docs.amd.com/en/develop/how-to/system-optimization/mi300x.html?highlight=IOMMU#iommu-configuration-systems-with-256-cpu-threads

https://rocm.docs.amd.com/en/develop/how-to/system-optimization/mi200.html?highlight=IOMMU#systems-with-256-cpu-threads-iommu-configuration

https://rocm.docs.amd.com/en/develop/how-to/system-optimization/mi100.html#systems-with-256-cpu-threads-iommu-configuration

Build TransferBench
---------------------

To build TransferBench using Makefile, use:

.. code-block:: bash

            $ make

To build TransferBench using CMake, use:

.. code-block:: bash

                $ mkdir build

                $ cd build

                $ CXX=/opt/rocm/bin/hipcc cmake ..

                $ make

.. note::

  If ROCm is installed in a folder other than ``/opt/rocm/``, set ``ROCM_PATH`` appropriately.

Build documentation
-----------------------

To build documentation locally, use:

.. code-block:: bash

  cd docs
  pip3 install -r .sphinx/requirements.txt
  python3 -m sphinx -T -E -b html -d _build/doctrees -D language=en . _build/html

> NVIDIA platform support

Reviewer: ```suggestion
NVIDIA CUDA platform support
```

Author: @lpaoletti  I am not sure about renaming the heading from NVIDIA platform support to NVIDIA CUDA platform support. In this section, we are explaining how users can build transferbench on NVIDIA hardware and that they need to install NVIDIA CUDA to achieve the same. The support is for NVIDIA hardware, so can we keep it just NVIDIA in the heading?

Reviewer: Makes sense

--------------------------

> You can build TransferBench to run on NVIDIA platforms via HIP or native NVCC.

Reviewer: ```suggestion
You can build TransferBench to run on NVIDIA CUDA platforms via HIP or native NVCC.
```

Reviewer: The HIP should be hyperlinked. 
You should also describe what is.

Author: Hyperlinked HIP

Author: This has been updated as per latest review from dev.

> To build with HIP for NVIDIA, install a HIP-compatible CUDA version such as CUDA 11.5 and use:

Reviewer: ```suggestion
To build with HIP for NVIDIA CUDA, install a HIP-compatible CUDA version such as CUDA 11.5 and use:
```

Reviewer: What are the HIP-compatible CUDA versions? make this a hyperlink

Author: https://rocm.docs.amd.com/projects/HIP/en/latest/how-to/faq.html#what-specific-version-of-cuda-does-hip-support mentions CUDA versions till 9.0 however Transferbench recommends CUDA 11.5. 
I am not sure if the HIP doc is updated. I will still verify with TransferBench dev as well.

Author: This has been removed as per latest updates from dev.
